### ISSUE TITLE
rvm can't install if user home directory contains @ symbol

### ISSUE THREAD

**Issue by sylvainsf** [rvm/rvm#3790]
## Description

If user's home directory contains an @ symbol rvm fails to install any rubies. 
## Environment info
- RVM version: 1.27.0
- Operating System: MacOS
- Version/Release:  10.12 Sierra
- Shell: zsh
## Expected behavior

rvm install ruby-2.2.0 should install ruby.
## Actual behavior

Note: User home directory is `/Users/sylvain.niles@alaskaair.com` but in output below you can see it's attempting to modify directories that do not exist. 

```
rvm install ruby-2.2.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.12/x86_64/ruby-2.2.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/sylvain.niles@alaskaair.com/.rvm/rubies/ruby-2.2.0, this may take a while depending on your cpu(s)...
ruby-2.2.0 - #downloading ruby-2.2.0, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12.6M  100 12.6M    0     0  2797k      0  0:00:04  0:00:04 --:--:-- 2909k
ruby-2.2.0 - #extracting ruby-2.2.0 to /Users/sylvain.niles@alaskaair.com/.rvm/src/ruby-2.2.0 - please wait
ruby-2.2.0 - #applying patch /Users/sylvain.niles@alaskaair.com/.rvm/patches/ruby/2.2.0/fix_installing_bundled_gems.patch - please wait
ruby-2.2.0 - #applying patch /Users/sylvain.niles@alaskaair.com/.rvm/patches/ruby/2.2.0/openssl3.patch - please wait
ruby-2.2.0 - #configuring - please wait
ruby-2.2.0 - #post-configuration - please wait
ruby-2.2.0 - #compiling - please wait
ruby-2.2.0 - #installing - please wait
ruby-2.2.0 - #making binaries executable - please wait
ruby-2.2.0 - #downloading rubygems-2.4.8
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  437k  100  437k    0     0   680k      0 --:--:-- --:--:-- --:--:--  679k
ruby-2.2.0 - #extracting rubygems-2.4.8 - please wait
ruby-2.2.0 - #removing old rubygems - please wait
ruby-2.2.0 - #installing rubygems-2.4.8 - please wait
mkdir: /Users/sylvain.niles@global/bin: Permission denied
mkdir: /Users/sylvain.niles@global/cache: Permission denied
ruby-2.2.0 - #gemset created /Users/sylvain.niles@global
ruby-2.2.0 is not installed.
To install do: 'rvm install ruby-2.2.0'
ruby-2.2.0 is not installed.
To install do: 'rvm install ruby-2.2.0'
```
## Steps to reproduce
1. Create user with home directory containing @ symbol
2. Install rvm
3. Attempt to install any version of ruby.

**Comment by sylvainsf** [comment]
@pkuczynski I'm not convinced it's macos specific, but don't have a linux instance to test on at the moment.

**Comment by sylvainsf** [comment]
If anyone can point me to where home directory is parsed/used I may be able to figure out a fix. I couldn't quite figure out where in the code on my own.

**Comment by Defilan** [comment]
I've run into this too. When using a company Mac that uses an e-mail address as the username, it breaks for me too. I was forced to use a non-compliant username to bypass this issue.

**Comment by Zipfer** [comment]
I have the same issue. How can I just workaround it?

**Comment by sylvainsf** [comment]
@Zipfer I just moved my home directory to a non @ symbol named directory but that was a big pain to do.

**Comment by nkirschbaum** [comment]
+1  I'm having the same "@" issue

**Comment by bradwilliamson** [comment]
I am having this issue on RHEL 7

**Comment by jeanlange** [comment]
Having this issue on MacOS High Sierra (10.13.4)

**Comment by Ramyashanthaa** [comment]
Iam using MacOS High Sierra (10.13.4) too. My username doesn't have "@".how to resolve the issue?

**Comment by pkuykendall** [comment]
I am also having this issue on RHEL 7.

**Comment by fborges** [comment]
Having this issue on MacOS 10.14.4, RVM 1.29.7

**Comment by tadas-s** [comment]
Having this issue on Ubuntu 18.04, RVM 1.29.8 .. User names in our developer machines include `@` sign between actual user name and domain user belongs to.

**Comment by islamaskar** [comment]
Sorry and update regarding this

**Comment by pkuczynski** [comment]
Not really. No time to work on this I am afraid :( But a PR is welcome

**Comment by skoblenick** [comment]
This also happens in Linux when using SSSD. The user's home directories are created as user@example.com. Installation of rvm fails with `mkdir: cannot create directory '/home/user@global': Permission denied`

**Comment by sourabhpat** [comment]
`mkdir: /Users/user.name@global/bin: Permission denied` 
`mkdir: /Users/user.name@global/bin: Permission denied` 

Any update on this? Facing same issue. Really need help

Thanks in advance.

**Comment by raelgc** [comment]
It appears to be affecting Ubuntu too: #4776.

**Comment by priyankshah217** [comment]
I am also facing the same issue.

**Comment by sourabhpat** [comment]
I have done these things and everything is working fine now.

Here is the solution for the same

```
cd /Users;
ln -s "<email_id / current username>" <new_name_home>; 
export HOME=/Users/new_name_home;
add to bash profile; 
curl command to install rvm; 
add .rvm scripts to path in bash_profile; 
source rvm; 
rvm install "ruby-2.5.5";
```

**Comment by MichaelRo** [comment]
Same problem here

**Comment by GroveIV** [comment]
Also having this problem

**Comment by jamesiarmes** [comment]
I believe I've identified the issue here. There are multiple places where rvm attempts to determine paths by removing the gemset from an existing path, if it exists. However, it blindly removes everything from the gemset separator (@) to the end of the string. I've opened #5221 which I can confirm allows rubies to be installed and gemsets to be created when your home directory includes the separator as a character. There may be more places where this logic needs to be updated, but this has addressed my immediate issue.

In the process of looking into this I attempted to override the separator, but this didn't work either. There are a number of places where the separator is hard coded to "@" rather than using the `rvm_gemset_separator` variable. I have not addressed this additional issue in the PR.
